### ISSUE TITLE
Global notificiations

### ISSUE THREAD

**Issue by briri** [DMPRoadmap/roadmap#754]
Create a page that allows Super Admins to create global system messages. 

Example use cases would be 'Funder template 'X' has been updated. Please review your customizations.' or 'System maintenance scheduled for 11/15/2017', etc.

Need to create a new page (accessed via the Super Admin menu) that:
- Lists all prior notifications
- Allows the user to add a new notification (text, start date, expiration date)
- Messages should appear on the user's 'My Dashboard' page at the top in the notifications section during the time window specified
- Allow the user to copy an old notification

**Comment by sjDCC** [comment]
We may want to adjust where the message display and to whom too. The main use cases I can think of for global notifications to be set by super admins are:
1. downtime announcements
2. new feature alerts
3. new content e.g. funder templates

Downtime announcements would definitely want to display to everyone and be visible on the homepage prior to login. Alerts about new content and features though should perhaps only present to certain groups or once logged in. The example given above of 'Funder template 'X' has been updated. Please review your customizations.' would be most pertinent to display to org admins rather than all users.

**Comment by stephaniesimms** [comment]
Thanks for revisiting @sjDCC. I agree that we would want to display:
1. downtime announcements on homepage, possibly also on My Dashboard
2. new feature alerts we handle through other newsfeeds currently (e.g., blog) - it might be nice to display them somewhere prominently in the UI as well but this isn't a requirement for DMPTool
3. changes that require users to take some action (e.g. funder template update) are the other main reason for global notifications. in this example, admins need to know about it so they can transfer customizations but users would also benefit from a notice about funder template updates. this is why we initially thought about global notifications having a central My Dashboard display. another way to handle notifications would be tucked into a nav bar as in your screenshot above, but i tend to ignore these. we could always test a prototype.

w/regard to mobile display, the site isn't responsive in its current form so i wouldn't let that affect design decisions for now.

overall, i favor keeping it simple and implementing one kind of/approach to global notifications

**Comment by sjDCC** [comment]
Thanks Steph. New feature alerts aren't a priority for us currently either - was just trying to flesh out when this may be used and for what kind of things as the mock-up leaves it pretty open. 

If we're really only talking about downtime announcements and new funder templates then perhaps we want them on the homepage rather than my dashboard? Downtime definitely wants to be public and possibly funder template news too.

Let's see what the French use cases are too and design from there.

**Comment by vyruss** [comment]
#### Notes from developer meeting of 6 Apr 18:
* Create a Notifications model class/table
    * `type` global (just this for now)
    * `alert` level i.e. urgent, info, etc for importance/colour
    * `title`: short title for the notification e.g. "Downtime: 3/12 08:00-05:00"
    * `text`: free text to be displayed upon click e.g. "On Friday 12 March from 8 to 5pm we will..."
    * `dismissable`/`closeable`: notifications you are able to close (e.g. you can't close downtime alerts)
    * `created_at` timestamp for ordering
    * `expires_at` timestamp to stop displaying it
* Acknowledgement table
    * keeps track of which user has dismissed (clicked on "Mark as seen") notification
* Implementation
    * Facebook-style notification icon with counter (of "unseen" notifications)
    * Github-style notifications page: (first row won't be visible)
         *  | Title | Timestamp | Seen | 
            | --- | --- | --- |
            | [New templates out for testing!](#) | 3/12/2018 08:35 | [Mark as seen](#) |
            | [Downtime: 3/12 08:00-05:00](#) | 3/12/2018 07:10 | ✓ |
    * Global notifications also appear in notification area as alerts
        * Non-dismissable notification alerts cannot be closed

**Comment by briri** [comment]
lets add a `starts_at` datetime field as well to the table so that admins can create the notification before they want it to display to users.

I'd also like to see what @benjaminfaure and his team have done from the UI perspective. A separate notifications page is a good long term strategy but a simple message area at the top of the page may be a good lightweight solution at this point (especially if we're really just focused on global outage type notifications right now).

**Comment by stephaniesimms** [comment]
looking forward to seeing what @benjaminfaure have done! just to summarize the requirements discussion between me and @sjDCC above, we're looking for a simple message area for a narrow set of global notifications (maintenance downtime, funder template changes). messages could be displayed on the homepage or My Dashboard.

**Comment by vyruss** [comment]
@stephaniesimms @briri agree, that's what I mean by "notification area" - i.e. where our alerts currently appear, where the "global" notifications will appear. The "notification page" is a separate page where one will also be able to see their "personal" notifications

**Comment by stephaniesimms** [comment]
ok thanks for clarifying @vyruss. there's nothing in the requirements for a personal notifications page so it sounds like we can focus on the global notifications.

**Comment by benjaminfaure** [comment]
If @Sudiukil is making good progress on the global notification, should we add a personal notifications page or should westop with the essential ?

**Comment by stephaniesimms** [comment]
we only need a global notifications page @benjaminfaure . no personal notifications. looking forward to the demo.

**Comment by vyruss** [comment]
@benjaminfaure not at this stage, I just included it in the notes as it was part of the conversation

**Comment by Sudiukil** [comment]
Quick question, how should we sort the notifications on the index page ? For now it's sorted by title, but I don't think it's the most natural sort order... what do you think ? Maybe by start date (with expiration date as second choice) ? Or simply by creation date (would require to add a column to the index though) ?

**Comment by vyruss** [comment]
@Sudiukil yes, I think start date, then expiration date makes sense (if we can sort by two fields).

**Comment by stephaniesimms** [comment]
PR in review. moving this issue to the board so we can test the UI and include it in next release.
